llvm/lib/Target/RISCV/RISCVVectorPeephole.cpp

```diff
@@ -585,6 +585,7 @@ bool RISCVVectorPeephole::foldUndefPassthruVMV_V_V(MachineInstr &MI) {
 
   MRI->replaceRegWith(MI.getOperand(0).getReg(), MI.getOperand(2).getReg());
   MI.eraseFromParent();
+  MRI->clearKillFlags(MI.getOperand(2).getReg());
```

Comment:
Same here, I think this should be
```suggestion
  MRI->replaceRegWith(MI.getOperand(0).getReg(), MI.getOperand(2).getReg());
  MRI->clearKillFlags(MI.getOperand(2).getReg());
  MI.eraseFromParent();
```